src/utils/media-utils.js

```diff
@@ -113,6 +117,18 @@ export const determineMediaDuration = memoizee(async url => {
   return Promise.race([playerPromise, timeoutPromise]);
 }, { promise: true });
 
+export const verifyYoutubeThumbnailUrl = memoizee(url => {
+  const imagePromise = new Promise((resolve, reject) => {
```

Comment:
I'm missing why this need to be wrapped in another promise.

Reply:
Thanks for spotting it, it's a leftover and is now removed.